src/migration/MigrationExecutor.ts

```diff
@@ -232,28 +261,99 @@ export class MigrationExecutor {
                         },
                         {
-                            name: "name",
+                            name: COLUMN_NAME,
                             type: this.connection.driver.normalizeType({type: this.connection.driver.mappedDataTypes.migrationName}),
                             isNullable: false
                         },
+                        {
+                            name: COLUMN_HASH,
+                            type: this.connection.driver.normalizeType({type: this.connection.driver.mappedDataTypes.migrationHash}),
+                            isNullable: false,
+                            length: "40"
+                        }
                     ]
                 },
             ));
         }
+        const table = await queryRunner.getTable(this.migrationsTable);
+        const hashColumnExist = table!.columns.some(c => c.name.toLowerCase() === COLUMN_HASH);
+        if (!hashColumnExist) {
+            await queryRunner.addColumn(this.migrationsTable, new TableColumn({
+                name: COLUMN_HASH,
+                type: this.connection.driver.normalizeType({type: this.connection.driver.mappedDataTypes.migrationHash}),
+                isNullable: true,
+                length: "40",
+            }));
+
+            await this.updateHashes(queryRunner);
+
+            const {count} = await this.connection.manager.createQueryBuilder(queryRunner)
+            .select("count(*)", "count")
+            .from(this.migrationsTable, this.migrationsTableName)
+            .where("hash is null")
+            .getRawOne();
+
+            if (count === 0) {
+                await queryRunner.changeColumn(this.migrationsTable, COLUMN_HASH, new TableColumn({
+                    name: COLUMN_HASH,
+                    type: this.connection.driver.normalizeType({type: this.connection.driver.mappedDataTypes.migrationHash}),
+                    isNullable: false,
+                    length: "40",
+                }));
+            }
+        }
+
+        const createDateColumnExist = table!.columns.some(c => c.name.toLowerCase() === COLUMN_CREATE_DATE);
+        if (!createDateColumnExist) {
+            await queryRunner.addColumn(this.migrationsTable, new TableColumn({
+                name: COLUMN_CREATE_DATE,
+                type: this.connection.driver.normalizeType({type: this.connection.driver.mappedDataTypes.createDate}),
+                default: this.connection.driver.mappedDataTypes.createDateDefault,
+                precision: this.connection.driver.mappedDataTypes.createDatePrecision,
+                isNullable: true,
+            }));
+        }
+    }
+
+    protected async updateHashes(queryRunner: QueryRunner): Promise<void> {
+        const executedMigrations = await this.loadExecutedMigrations(queryRunner);
+        const migrations = this.getMigrations().filter(migration => executedMigrations.find(executedMigration => executedMigration.name === migration.name));
```

Comment:
Is it me or is that line hard to read? You're looking for migrations which wasn't executed? It should be possible to rewrite it to more readeable format or just use more precise name - you're not selecting all migrations, just portion of them.